packages/slider/src/ux-slider.ts

```diff
@@ -0,0 +1,158 @@
+import { customElement, bindable } from 'aurelia-templating';
+import { inject } from 'aurelia-dependency-injection';
+import { StyleEngine, UxComponent } from '@aurelia-ux/core';
+import { UxSliderTheme } from './ux-slider-theme';
+import { computedFrom, bindingMode } from 'aurelia-binding';
+
+@inject(Element, StyleEngine)
+@customElement('ux-slider')
+export class UxSlider implements UxComponent {
+  private isActive: boolean;
+  private percentValue: number;
+  private onMouseMove: (e: MouseEvent) => void = (e) => this.updateValue(e.clientX);
+  private onMouseUp: (e: MouseEvent) => void = this.handleMouseUp.bind(this);
+
+  @bindable public theme: UxSliderTheme;
+  @bindable({ defaultBindingMode: bindingMode.twoWay }) public value: number;
+  @bindable public min: number;
+  @bindable public max: number;
+  @bindable public disabled: boolean;
+  @bindable public step: number;
+
+  constructor(
+    public element: HTMLElement,
+    private styleEngine: StyleEngine) { }
+
+  @computedFrom('percentValue')
+  get sliderBeforeWidth(): number {
+    return this.percentValue * 100;
+  }
+
+  @computedFrom('percentValue')
+  get sliderAfterWidth(): number {
+    return (1 - this.percentValue) * 100;
+  }
+
+  public bind() {
+    this.themeChanged(this.theme);
+
+    this.minChanged();
+    this.maxChanged();
+    this.valueChanged();
+    this.disabledChanged();
+    this.stepChanged();
+  }
+
+  public detached() {
+    window.removeEventListener('mouseup', this.onMouseUp);
```

Comment:
Maybe we can use `ElementsEvent` from `aurelia-templating` to make it simpler? It's used in some other places in UX too I think. Or at least we should use `handleEvent`.

Reply:
updated to use ElementEvents similar to how it's used in the checkbox control

Reply:
Very nice : 💓